hknweb/academics/models/icsr.py

```diff
@@ -8,15 +9,51 @@ class ICSR(AcademicEntity):
     InstructorCourseSemesterRelation (ICSR)
     """
     # reference attributes
-    icsr_course = models.ForeignKey('Course', on_delete=models.PROTECT, related_name='icsr_course')
-    icsr_department = models.ForeignKey('Department', on_delete=models.PROTECT, related_name='icsr_department')
-    icsr_instructor = models.ForeignKey('Instructor', on_delete=models.PROTECT, related_name='icsr_instructor')
-    icsr_semester = models.ForeignKey('Semester', on_delete=models.PROTECT, related_name='icsr_semester')
+    icsr_course = models.ForeignKey('Course', on_delete=models.PROTECT)
+    icsr_department = models.ForeignKey('Department', on_delete=models.PROTECT)
+    icsr_instructor = models.ForeignKey('Instructor', on_delete=models.PROTECT)
+    icsr_semester = models.ForeignKey('Semester', on_delete=models.PROTECT)
 
     # value attributes
     first_name = models.TextField(max_length=100)
     last_name = models.TextField(max_length=100)
     course_number = models.TextField(max_length=100)
+    course_name = models.TextField(max_length=200)
     section_type = models.TextField(max_length=100)
     section_number = models.TextField(max_length=100)
     instructor_type = models.TextField(max_length=100)
+
+    @staticmethod
+    def recency_ordering(qs):
+        from .logistics import Semester
+        return qs.annotate(numerical_semester=Case(
+            When(icsr_semester__year_section=Semester.SPRING, then=Value(1)),
+            When(icsr_semester__year_section=Semester.SUMMER, then=Value(2)),
+            When(icsr_semester__year_section=Semester.FALL, then=Value(3)))).order_by \
+            ("-icsr_semester__year", "-numerical_semester")
+
+    def save(self, *args, **kwargs):

```

Comment:
@Rishabhk2357 and I synced offline -- the plan is to generalize this `save` functionality to `AcademicEntity` and also introduce appropriate `pre_delete`/`post_delete` actions.